### DISCUSSION TITLE
Why Rust dependency?

### DISCUSSION THREAD

**Discussion by SirZenith** [OscarCreator/rsync.nvim#121] [Q&A] [answered]
Could it be better if a pure Lua TOML library is used?

As I observed from the code, external Rust library is used and only used for parsing TOML file.

Having to build Rust library make it not so friendly for portability. 

On Windows, one can build their own `luajit.lib` and manually change mlua version in `Cargo.toml` to the latest, then every thing builds. Not too big of  a problem, if one is willing to pay this extra effort.

But I just gave up thinking when it comes to Android.

There are Lua implementation for TOML parsing, so I think maybe use one of those can make thinkings a little easier.

**Comment by OscarCreator** [comment, marked as answer]
The reason is quite simple, in the environment where I use this the Lua toml library is blocked for download.
